frontends/verilog/preproc.cc

```diff
@@ -495,6 +511,10 @@ static bool try_expand_macro(define_map_t &defines, std::string &tok)
 			args.push_back(arg);
 		}
 		for (const auto &pr : body->args.get_vals(name, args)) {
+			if (const define_body_t *existing = defines.find(pr.first)) {
+				macro_arg_stack.push({pr.first, *existing});
+				insert_input("`__restore_macro_arg");
```

Comment:
this needs armoring with spaces, otherwise:

```
module top(...);

`define add(x) x +

input [3:0] A;
output [3:0] B;
assign B = `add(`add(3)A)A;

endmodule
```

-->

```
-- Verilog code after preprocessor --
`file_push "t.v"
module top(...);



input [3:0] A;
output [3:0] B;
assign B = 3 +`__restore_macro_argA +A;

endmodule

`file_pop
-- END OF DUMP --
```

This, of course, blows up.

Reply:
Nice catch, thanks!